yunikorn-web.yaml

```diff
@@ -0,0 +1,57 @@
+package:
+  name: yunikorn-web
+  version: 1.6.0
+  epoch: 0
+  description: Apache YuniKorn Web UI
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - node-gyp
+      - nodejs-18
+      - npm
+      - pnpm
+      - python3
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: a51cc7203533f4134f3c1ea32a714f33876217f6
+      repository: https://github.com/apache/yunikorn-web
+      tag: v${{package.version}}
+
+  - uses: patch
+    with:
+      patches: |
+        0001-increase-the-budget.patch
+
+  - runs: |
+      mkdir -p ${{targets.contextdir}}/html
+      PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1 pnpm i
+      pnpm build:prod
+      cp -r dist/yunikorn-web/* ${{targets.contextdir}}/html/
+
+  - uses: go/build
+    with:
+      output: yunikorn-web
+      packages: ./pkg/cmd/web/
+      modroot: .
+      ldflags: |
+        -X main.version=${{package.version}}
+        -X main.date=$(date -u -d "@${SOURCE_DATE_EPOCH:-$(date +%s)}" "+%Y-%m-%dT%H:%M:%SZ")
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: "Compatibility package to place binaries in the location expected by upstream Dockerfile"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/
+          ln -sf /usr/bin/yunikorn-web ${{targets.contextdir}}/yunikorn-web
+
+update:
+  enabled: true
+  github:
+    identifier: apache/yunikorn-web
+    strip-prefix: v
```

Comment:
Since this seems to be a go binary that serves the html files, let's add a test to validate that they're being embedded or served correctly.